File: src/main/java/practice/payment/domain/Payment.java
package practice.payment.domain;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.experimental.FieldDefaults;
import practice.user.domain.User;

@Getter
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE)
public class Payment {

  User user;
  Long partnerId;
  Long paymentId;
  String partnerPayKey;
  List<Product> productItems;
  BigDecimal totalPayAmount;
  LocalDateTime paymentDateTime;

}

File: src/main/kotlin/practice/payment/domain/Payment.kt
package practice.payment.domain

import practice.user.domain.User
import java.math.BigDecimal
import java.time.LocalDateTime

class Payment(
    val user: User,
    val partnerId: Long,
    val paymentId: Long? = null,
    val partnerPayKey: String,
    val productItems: List<Product>,
    val totalPayAmount: BigDecimal,
    val paymentDateTime: LocalDateTime? = null,
)
